lib/qrCode.js

```diff
@@ -0,0 +1,55 @@
+const fetch = require("node-fetch");
+const crypto = require('crypto');
+
+const baseUrl = "https://yayawallet.com/api/en";
+
+async function getTime(){
+  const url = `${baseUrl}/time`
+  const response = await fetch(url, {
+    method: 'GET',
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  const unixResponse = await response.json();
+  return unixResponse.time;
+}
+
+function generateSignature(timestamp, method, endpoint, body, apiSecret) {
```

Comment:
should not this go to some common place where each business logic reuses to set the request options including the HTTP headers?

All request will need the signature, so better put it outside the QR code generation logic, in a reusable way for other business logics.